controllers/courses_controller.js

```diff
@@ -3,14 +3,35 @@ const sql = require('../sql');
 const log = require('../helpers/logging');
 
 exports.index = (req, res, next) => {
-  db.query(sql.courses.queries.get_courses, (err, data) => {
-    if (err) {
-      log.error('Failed to get courses');
-      next(err);
-    } else {
-      res.render('courses', { data: data.rows });
-    }
-  });
+  const { query } = req;
```

Comment:
Destructure here to obtain both `semester_name` and `module_code` then proceed to check if they are `undefined`